Code/Editor/EditorViewportWidget.cpp

```diff
@@ -2562,130 +2287,126 @@ void EditorViewportWidget::SetFOV(float fov)
+        return fov;
+    }
+    else
     {
-        CBaseObject* cameraObject = GetCameraObject();
+        return AZ::GetPerspectiveMatrixFOV(m_defaultView->GetViewToClipMatrix());
+    }
+}
 
-        AZ::EntityId cameraEntityId;
-        AzToolsFramework::ComponentEntityObjectRequestBus::EventResult(cameraEntityId, cameraObject, &AzToolsFramework::ComponentEntityObjectRequestBus::Events::GetAssociatedEntityId);
-        if (cameraEntityId.IsValid())
-        {
-            // component Camera
-            float fov = DEFAULT_FOV;
-            Camera::CameraRequestBus::EventResult(fov, cameraEntityId, &Camera::CameraComponentRequests::GetFov);
-            return AZ::DegToRad(fov);
-        }
+void EditorViewportWidget::OnActiveViewChanged(const AZ::EntityId& viewEntityId)
+{
+    // Avoid re-entry
+    if (m_sendingOnActiveChanged)
+    {
+        return;
     }
 
-    if (m_viewEntityId.IsValid())
+    // Ignore any changes in simulation mode
+    if (m_playInEditorState != PlayInEditorState::Editor)
     {
-        float fov = AZ::RadToDeg(m_camFOV);
-        Camera::CameraRequestBus::EventResult(fov, m_viewEntityId, &Camera::CameraComponentRequests::GetFov);
-        return AZ::DegToRad(fov);
+        return;
     }
 
-    return m_camFOV;
-}
+    // if they've picked the same camera, then that means they want to toggle
+    if (viewEntityId.IsValid())
+    {
+        // Any such events for game entities should be filtered out by the check above
+        AZ_Error(
+            "EditorViewportWidget",
+            Camera::EditorCameraViewRequestBus::FindFirstHandler(viewEntityId) != nullptr,
+            "Internal logic error - active view changed to an entity which is not an editor camera. "
+            "Please report this as a bug."
+        );
 
-//////////////////////////////////////////////////////////////////////////
-bool EditorViewportWidget::CreateRenderContext()
-{
-    return true;
-}
+        m_viewEntityId = viewEntityId;
+        m_viewSourceType = ViewSourceType::CameraComponent;
+        AZStd::string entityName;
+        AZ::ComponentApplicationBus::BroadcastResult(entityName, &AZ::ComponentApplicationRequests::GetEntityName, viewEntityId);
+        SetName(QString("Camera entity: %1").arg(entityName.c_str()));
 
-//////////////////////////////////////////////////////////////////////////
-void EditorViewportWidget::DestroyRenderContext()
-{
+        PostCameraSet();
+    }
+    else
+    {
+        SetDefaultCamera();
+    }
 }
 
 //////////////////////////////////////////////////////////////////////////
 void EditorViewportWidget::SetDefaultCamera()
 {
-    if (IsDefaultCamera())
-    {
-        return;
-    }
-    ResetToViewSourceType(ViewSourceType::None);
-    GetViewManager()->SetCameraObjectId(m_cameraObjectId);
+    m_viewEntityId.SetInvalid();
+    m_viewSourceType = ViewSourceType::None;
+    GetViewManager()->SetCameraObjectId(GUID_NULL);
```

Comment:
comment: We also aiming to remove the need for ViewManager too (the new Atom Viewport code makes this redundant) It'll be awesome when we can remove all that